src/GraphQLParser/Visitors/SDLWriter.cs

```diff
@@ -0,0 +1,765 @@
+                        await context.Writer.WriteAsync(" | ");
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitDirective(GraphQLDirective directive, TContext context)
+        {
+            await Visit(directive.Comment, context);
+            await context.Writer.WriteAsync('@');
+            await Visit(directive.Name, context);
+            if (directive.Arguments != null)
+            {
+                await context.Writer.WriteAsync('(');
+                await VisitArguments(directive, context);
+                await context.Writer.WriteAsync(')');
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitArgument(GraphQLArgument argument, TContext context)
+        {
+            await Visit(argument.Comment, context);
+            await Visit(argument.Name, context);
+            await context.Writer.WriteAsync(": ");
+            await Visit(argument.Value, context);
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitNonNullType(GraphQLNonNullType nonNullType, TContext context)
+        {
+            await Visit(nonNullType.Comment, context);
+            await Visit(nonNullType.Type, context);
+            await context.Writer.WriteAsync('!');
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitListType(GraphQLListType listType, TContext context)
+        {
+            await Visit(listType.Comment, context);
+            await context.Writer.WriteAsync('[');
+            await Visit(listType.Type, context);
+            await context.Writer.WriteAsync(']');
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitListValue(GraphQLListValue listValue, TContext context)
+        {
+            await Visit(listValue.Comment, context);
+            if (listValue.Values?.Count > 0)
+            {
+                await context.Writer.WriteAsync('[');
+                for (int i=0; i<listValue.Values.Count; ++i)
+                {
+                    await Visit(listValue.Values[i], context);
+                    if (i < listValue.Values.Count - 1)
+                        await context.Writer.WriteAsync(", ");
+                }
+                await context.Writer.WriteAsync(']');
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitNullValue(GraphQLScalarValue nullValue, TContext context)
+        {
+            await Visit(nullValue.Comment, context);
+            await context.Writer.WriteAsync("null");
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask VisitStringValue(GraphQLScalarValue stringValue, TContext context)
+        {
+            await Visit(stringValue.Comment, context);
+            await context.Writer.WriteAsync('\"');
+            await Write(context, stringValue.Value);
```

Comment:
Needs escaping of various characters; please add test.

Optional: detect if the string is multi-line, and can be formatted as a blockstring without loss of data, and if both are true, print it that way.

Reply:
> and can be formatted as a blockstring without loss of data

Example ?

Reply:
I took test data from lexer tests. This is your test with 30 cases. I changed some values. Honestly, work with escape symbols is very tired. I would like to finish with this PR and go on. I could be mistaken somewhere, further use will show. We will add tests. Or you can add additional test cases yourself where you think right.

Reply:
I agree to finish the PR and go on.  The API will not change if we improve these methods in a later PR, either to fix an issue or enhance performance.

Reply:
I'm OK to wait 1-2-3 days for your review. At least until I start a new branch working on type extensions. I am 80% sure that I have errors in formatting/edge cases.